core/src/main/java/org/fao/geonet/util/XslUtil.java

```diff
@@ -702,4 +707,57 @@ public static String getMax(Object values) {
         }
         return max;
     }
+
+    private static final Cache<String, Boolean> URL_VALIDATION_CACHE;
+
+    static {
+        URL_VALIDATION_CACHE = CacheBuilder.<String, Boolean>newBuilder().
+                maximumSize(100000).
+                expireAfterAccess(25, TimeUnit.HOURS).
+                build();
+    }
+
+    public static boolean validateURL(final String urlString) throws ExecutionException {
+        return URL_VALIDATION_CACHE.get(urlString, new Callable<Boolean>() {
+            @Override
+            public Boolean call() throws Exception {
+                try {
+                    return validateURLImpl(urlString, 1);
+                } catch (Exception e) {
+                    return false;
+                }
+            }
+        });
+    }
+
+    private static boolean validateURLImpl(final String urlString, int tryNumber) throws IOException {
```

Comment:
Maybe better to merge with getUrlStatus https://github.com/geonetwork/core-geonetwork/blob/develop/core/src/main/java/org/fao/geonet/util/XslUtil.java#L441-L505 ?
 

Current usage: https://github.com/geonetwork/core-geonetwork/blob/develop/schemas/iso19139/src/main/plugin/iso19139/process/linked-data-checker.xsl#L67